cfme/cloud/flavor.py

```diff
@@ -98,6 +104,23 @@ def is_displayed(self):
     entities = FlavorDetailsEntities()
 
 
+class FlavorAddView(BaseLoggedInPage):
+    """ Represents Add NetworkRouters page """
+    provider = Select(name='ems_id')
+    flavor_name = TextInput(name='name')
+    ram_size = TextInput(name='ram')
+    vcpus = TextInput(name='vcpus')
+    disk_size = TextInput(name='disk')
+    swap_size = TextInput(name='swap')
+    rxtx_factor = TextInput(name='rxtx_factor')
+    public = BootstrapSwitch(name='is_public')
+    add = Button('Add')
+
+    @property
+    def is_displayed(self):
```

Comment:
Is_displayed should check if the view is displayed and return True or False based on that .
Example : 
https://github.com/ManageIQ/integration_tests/blob/master/cfme/services/catalogs/catalog_items/ansible_catalog_items.py#L200